hw/drivers/sensors/bmp388/src/bmp388.c

```diff
@@ -0,0 +1,4138 @@
+			/* Multiply with pressure or temperature header len */
+			*watermark_len = dev->fifo->data.req_frames * BMP3_P_OR_T_HEADER_DATA_LEN;
+		} else {
+			/* No sensor is enabled */
+			rslt = BMP3_W_SENSOR_NOT_ENABLED;
+		}
+	} else {
+		/* Required frame count is zero, which is invalid */
+		rslt = BMP3_W_INVALID_FIFO_REQ_FRAME_CNT;
+	}
+
+	return rslt;
+}
+
+/*!
+ * @brief This API sets the fifo watermark length according to the frames count
+ * set by the user in the device structure. Refer below for usage.
+ */
+int8_t bmp3_set_fifo_watermark(struct sensor_itf *itf, const struct bmp3_dev *dev)
+{
+	int8_t rslt;
+	uint8_t reg_data[2];
+	uint8_t reg_addr[2] = {BMP3_FIFO_WM_ADDR, BMP3_FIFO_WM_ADDR+1};
+	uint16_t watermark_len;
+
+	rslt = null_ptr_check(dev);
+
+	if ((rslt == BMP3_OK) && (dev->fifo != NULL)) {
+		rslt = convert_frames_to_bytes(&watermark_len, dev);
+		if (rslt == BMP3_OK) {
+			reg_data[0] = BMP3_GET_LSB(watermark_len);
+			reg_data[1] = BMP3_GET_MSB(watermark_len) & 0x01;
+			rslt = bmp3_set_regs(itf, reg_addr, reg_data, 2, dev);
+		}
+	}
+
+	return rslt;
+}
+
+int8_t bmp388_configure_fifo_with_watermark(struct sensor_itf *itf, struct bmp3_dev *dev, uint8_t en)
+{
+	int8_t rslt;
+	/* FIFO object to be assigned to device structure */
+	struct bmp3_fifo fifo;
+    /* Used to select the settings user needs to change */
+	uint16_t settings_sel;
+
+	/* Enable fifo */
+	fifo.settings.mode = BMP3_ENABLE;
+	/* Enable Pressure sensor for fifo */
+	fifo.settings.press_en = BMP3_ENABLE;
+	/* Enable temperature sensor for fifo */
+	fifo.settings.temp_en = BMP3_ENABLE;
+	/* Enable fifo time */
+	fifo.settings.time_en = BMP3_ENABLE;
+	/* No subsampling for FIFO */
+	fifo.settings.down_sampling = BMP3_FIFO_NO_SUBSAMPLING;
+	fifo.settings.fwtm_en = en;
+	/* Link the fifo object to device structure */
+	dev->fifo = &fifo;
+	/* Select the settings required for fifo */
+	settings_sel = BMP3_FIFO_MODE_SEL | BMP3_FIFO_TIME_EN_SEL | BMP3_FIFO_TEMP_EN_SEL |
+	    BMP3_FIFO_PRESS_EN_SEL | BMP3_FIFO_DOWN_SAMPLING_SEL | BMP3_FIFO_FWTM_EN_SEL;
+    /* Set the selected settings in fifo */
+	rslt = bmp3_set_fifo_settings(itf, settings_sel, dev);
+    if (rslt != 0) {
+	BMP388_LOG(ERROR, "bmp3_set_fifo_settings failed %d\n", rslt);
+	goto ERR;
+    }
+	/* Set the number of frames to be read so as to set the watermark length in the sensor */
+	//dev->fifo->data.req_frames = 50;
+	dev->fifo->data.req_frames = g_bmp388_dev.fifo_watermark_level;
+	rslt = bmp3_set_fifo_watermark(itf, dev);
+	if (rslt != 0) {
+	BMP388_LOG(ERROR, "bmp3_set_fifo_watermark failed %d\n", rslt);
+	goto ERR;
+    }
+ERR:
+	return rslt;
+
+}
+
+int8_t bmp388_configure_fifo_with_fifofull(struct sensor_itf *itf, struct bmp3_dev *dev, uint8_t en)
+{
+	int8_t rslt;
+	/* FIFO object to be assigned to device structure */
+	struct bmp3_fifo fifo;
+    /* Used to select the settings user needs to change */
+	uint16_t settings_sel;
+
+	/* Enable fifo */
+	fifo.settings.mode = BMP3_EN
```

Comment:
Please use 4 spaces instead of tabs for indentation for all lines of code. It would be nice if you could update bma253 driver as well.